### ISSUE TITLE
Felix build is failing which impacts Calico build

### ISSUE THREAD

**Issue by chakravarthy392** [projectcalico/felix#1864]
Felix build is getting failed on Intel and non-Intel architectures also.

## Expected Behavior
`Calico-felix` binary and felix docker image should get created.

## Current Behavior
Build failed with below message.
`make: *** [bin/calico-felix] Error 2`
`Makefile:334: recipe for target 'bin/calico-felix' failed`

## Steps to Reproduce (for bugs)

1. wget https://storage.googleapis.com/golang/go1.10.1.linux-amd64.tar.gz
2. tar xf go1.10.1.linux-amd64.tar.gz
3. export GOPATH=$PWD
4. export GOROOT=$PWD/go
5. export PATH=$GOROOT/bin:$PATH
6. git clone https://github.com/projectcalico/felix $GOPATH/src/github.com/projectcalico/felix
7. cd $GOPATH/src/github.com/projectcalico/felix
8. git checkout 3.0.7
9. make calico/felix

## Context
Pushed `calico/felix:3.0.7` image is getting downloaded for Intel. But if we try to build it from source, docker image will not be created [ [X86_felix_build.log.txt](https://github.com/projectcalico/felix/files/2209255/X86_felix_build.log.txt)] and this issue becomes a blocker for building Calico node.
It is a severe issue on non-Intel architectures.

## Your Environment
* Calico version - v3.0.8
* Felix version - v3.0.7
* Operating System and version: Intel - Ubuntu16.04

**Comment by fasaxc** [comment]
Looks like a bad build of the protoc image.  Will check who pushed that and see if we can back it out

**Comment by chakravarthy392** [comment]
Felix is getting built successfully with protobuf_3.5.1 old image on Intel, but the problem remains same if it gets overwritten with recently built protobuf and the interesting thing is that 3.5.1 is only the recent tag which is being used in the latest build!!!
So here issue may not be related to version of Protobuf.

**Comment by basheer12** [comment]
Facing similar issue in building the felix image on ppc64le arch.

## Environment:
ubuntu16.04
calico version- v3.1.3
Felix version - 3.1.3

## Error:
```
proto/felixbackend.pb.go:116:33: undefined: "github.com/projectcalico/felix/vendor/github.com/gogo/protobuf/proto".InternalMessageInfo
proto/felixbackend.pb.go:180:33: undefined: "github.com/projectcalico/felix/vendor/github.com/gogo/protobuf/proto".InternalMessageInfo
proto/felixbackend.pb.go:914:35: undefined: "github.com/projectcalico/felix/vendor/github.com/gogo/protobuf/proto".InternalMessageInfo
```

**Comment by chakravarthy392** [comment]
@basheer12 yes, the issue is not related to any specific arch.

**Comment by jsrz** [comment]
I'm experiencing the same issue on Arm64 with release 3.1.3.

**Comment by chakravarthy392** [comment]
@basheer12 .@jsrz While building protoc image it is forking golang generators from [protobuf](https://github.com/gogo/protobuf) , I have observed that some changes are done to that repo recently , I have used older version(v1.0.0) of golang generators, protobuf image has got created but getting below error while building felix by using that protoc image.
`felixbackend.proto: No such file or directory`.
 with the latest version v1.1.x and master we are getting `felixbackend.pb.go` issue anyway.
